src/language-usability.rst

```diff
@@ -16,6 +16,171 @@ Usability of Type Systems
+    :cite:`myers2016improving`.
+
+
+API Design Aspects
+------------------
+
+..
+  Outline
+  * Many aspects to design
+  * Also tradeoffs between aspects.
+    * Broad tradeoff - learnability and complexity
+  * Paper: Improving API usability work
+  * Paper: Automated measure paper
+..
+
+There are several design aspects to desiging usable APIs.
+A broad aspect of usability identified by both :cite:`myers2016improving`
+and :cite:`scheller2015automated` is the effect of *complexity* on usability.
+Changing the inherant complexity exhibits one of the many tradeoffs between design aspects of APIs.
+Making an API more powerful (allowing a user to do more things with the API)
+by increasing complexity decreases the learnability of the API.
+However, increasing the learnability of an API by lowering the complexity also lowers the power of the API.
+
+.. container:: bib-item
+
+  .. bibliography:: api-usability.bib
+    :filter: key == 'myers2016improving'
+
+  This paper provides an overview of the design aspects of API design
+  (Learnability, Productivity, Error-Prevention, Simplicity, Consistency, and Matching Mental Models)
+  and the stakeholders (API designer, API user, and customer) involved with each design aspect.
+  This paper provides a high level survey on human-centered methods for designing APIs,
+  evaluating APIs, mitigating problems in APIs, and tools for improving API usability.
+
+.. container:: bib-item
+
+  .. bibliography:: api-usability.bib
+    :filter: key == 'scheller2015automated'
+
+  This paper identifies key aspects in API usability, in an *API Concepts Framework*,
+  a structure where *concepts* represent possible user actions, leading
+  to measureable properties on usability. Using this framework, the paper introduces
+  an automated method for measuring API usability, which they test in a user study.
+
+User Studies
+------------
+
+..
+  Outline
+  * User study studying global usability of APIs very difficult, so many factors
+  * Most user studies in API examine a specific aspect of api design and test it
+  * Paper: Usability Implications of Requiring Parameters in Objects' Constructors
+  * Paper: The Implications of Method Placement on API Learnability
+..
+
+Testing how to best design APIs is immensely difficult, as the numerous API design
+decisions all affect the overall usability interdependently
+:cite:`myers2016improving`. Testing the high-level and global design decisions
+is often intractable because of this. Most user studies in API usability
+examine a specific aspect of API design :cite:`stylos2007usability`
+:cite:`stylos2008implications`.
+
+.. container:: bib-item
+
+  .. bibliography:: api-usability.bib
+    :filter: key == 'stylos2007usability'
+
+  This paper examines the impact of requiring parameters in
+  objects' constructors on usability. The study in this paper showed that developers
```

Comment:
-> "This paper examines the usability implications of class constructor parameters."